bin/flutter

```diff
@@ -102,6 +102,12 @@ function upgrade_flutter () {
   fi
 
   local revision="$(cd "$FLUTTER_ROOT"; git rev-parse HEAD)"
+
+  # Invalidate cache if:
```

Comment:
add some documentation, for those a little rusty on their bash file test operators